### DISCUSSION TITLE
What are the limitations of Bun compare to Node and Deno?

### DISCUSSION THREAD

**Discussion by satyamskillz** [oven-sh/bun#729] [Q&A] [answered]
First of all, I love this framework. 

I was starting a new project and looking for something fast and reliable. Since this is a new framework, I didn't find many articles and projects that demonstrate the capabilities of Bun. 

I am very curious about its benefits and especially its limitations.

**Comment by xhyrom** [comment, marked as answer]
Benefits:
- Fast
- Performance
- All in one (bundler, transpiler, runtime, package manager)
- Written in Zig

Limitations:
- Not every nodejs native module is implemented (https://github.com/oven-sh/bun/issues/159)
- Many issues (https://github.com/oven-sh/bun/issues)
- Windows Support but you can use WSL (https://github.com/oven-sh/bun/issues/709)
- Hard to contribute for now (https://github.com/oven-sh/bun#developing-bun)
  ^ https://github.com/oven-sh/bun-dependencies is planned

**TL;DR** You can see that the bun is still new. Bun is moving forward quickly, but it's going to take a while. But you can definitely create a project that will work perfectly.

**Reply by satyamskillz** [reply]
I am very excited for the upcoming NextJS full support. Thanks for Bun ✨

**Reply by j-joker** [reply]
It puzzles me why Bun hasn't prioritized supporting Next.js sooner to attract a vast user base.
